### PR TITLE
update docs/sources/git.md

### PR DESCRIPTION
**Pull request by akihikokuroda**
Part of #522 

Signed-off-by: akihikokuroda <akuroda@us.ibm.com>

### CODE REVIEW

File: docs/sources/git.md [hunk 1]
```diff
@@ -1,29 +1,86 @@
...
+      ref:
+        branch: main
+      repository: https://github.com/operator-framework/combo
+  provisionerClassName: core-rukpak-io-plain
+```
 
-When creating a Bundle from a git source, a reference to a particular commit, tag, or branch must be provided for the
-provisioner to know where the bundle content is stored in the repository. Only one can be specified, and it is expected
-that the manifests are present in the particular commit/tag/branch at the directory specified.
+Referencing a different contents directory than the default:
```
Comment (tylerslaton) [inline docs/sources/git.md:63]:
```suggestion
Referencing a different content directory than the default:
```

File: docs/sources/git.md [hunk 2]
```diff
@@ -1,29 +1,86 @@
...
+in the particular commit/tag/branch at the directory specified.
 
-For example, for a plain+v0 bundle, the git repository backing the bundle, or "bundle git repo", must have a certain
-structure in order to produce a valid Bundle that works with the plain provisioner. It should have a manifests directory
-where the Kubernetes manifests are stored -- this directory is used when copying manifests onto the cluster. By default,
-the `manifests` directory is assumed to be at the root level, but a different bundle root can be specified
-via `spec.source.git.directory`.
+## Examples
+
+Referencing a git repository by a commit:
```
Comment (tylerslaton) [inline docs/sources/git.md:12]: Can we make these be `###` tags to make the document flow a bit better? Applies to all subsequently similar lines.

```suggestion
### Referencing a git repository by a commit
```

File: docs/sources/git.md [hunk 3]
```diff
@@ -1,29 +1,86 @@
...
+The git source provides the contents in a git repository as the source of the bundle.  The `source.type` for the git source is `git`.
+When creating a git source, a reference to a particular commit, tag, or branch must be provided in addition to the URL of the
+repository and an optional directory in the repository. It is expected that a proper format of bundle content is present
+in the particular commit/tag/branch at the directory specified.
 
-For example, for a plain+v0 bundle, the git repository backing the bundle, or "bundle git repo", must have a certain
-structure in order to produce a valid Bundle that works with the plain provisioner. It should have a manifests directory
-where the Kubernetes manifests are stored -- this directory is used when copying manifests onto the cluster. By default,
-the `manifests` directory is assumed to be at the root level, but a different bundle root can be specified
-via `spec.source.git.directory`.
```
Comment (tylerslaton) [inline docs/sources/git.md:13]: Is it necessary to remove this description?

File: docs/sources/git.md [hunk 4]
```diff
@@ -1,29 +1,86 @@
...
+        branch: main
+      directory: ./dev/deploy
+      repository: https://github.com/exdx/combo-bundle
+  provisionerClassName: core-rukpak-io-plain
+```
 
 ## Private git repositories
 
-A Bundle can reference content in a private git repository by creating a secret in the namespace that the provisioner is deployed.
+A git source can provide the contents in a private git repository by creating a secret in the namespace that the provisioner is deployed.
```
Comment (tylerslaton) [inline docs/sources/git.md:83]:
```suggestion
A git source can reference content in a private git repository by creating a secret in the namespace that the provisioner is deployed.
```

**Review by tylerslaton — COMMENTED** [review]
Some specific suggestions. Could you help me understand why we're removing content in some parts of this document?

**Comment by akihikokuroda** [comment]
@tylerslaton Thanks for review.  I removed some contents because this is just for the git source.  The contents about the bundle or provisioner should be put into docs/bundles/xxx.md or docs/provisioners/xxx.md.

**Comment by tylerslaton** [comment]
> @tylerslaton Thanks for review. I removed some contents because this is just for the git source. The contents about the bundle or provisioner should be put into docs/bundles/xxx.md or docs/provisioners/xxx.md.

Awesome, thanks for explaining that.

File: docs/sources/git.md (hunk 3 above)
Comment (akihikokuroda) [reply docs/sources/git.md:13]: Yes, this is about the `plain-v0` format.  It should go in `docs/bundles/plain.md`

Comment (tylerslaton) [reply docs/sources/git.md:13]: Ah, right. Thank you.

**Review by tylerslaton — APPROVED** [review]

**Review by joelanford — APPROVED** [review]

**Pull request merged**